src/Microphysics/microphysics_diagnostics.jl

```diff
@@ -0,0 +1,149 @@
+model = AtmosphereModel(grid)
+set!(model, θ=300)
+qᵛ⁺ = SaturationSpecificHumidityField(model, :prognostic)
+```
+
+Equilibrium flavor
+
+```@example ssh
+qᵛ⁺ₑ = SaturationSpecificHumidityField(model, :equilibrium)
+```
+
+Equilibrium flavor
+
+```@example ssh
+qᵛ = SaturationSpecificHumidityField(model, :total_moisture)
+```
+"""
+function SaturationSpecificHumidityField(model, flavor_symbol=:prognostic)
+
+    flavor = if flavor_symbol == :prognostic
+        Prognostic()
+    elseif flavor_symbol == :equilibrium
+        Equilibrium()
+    elseif flavor_symbol == :total_moisture
+        TotalMoisture()
+    else
+        valid_flavors = (:prognostic, :equilibrium, :total_moisture)
+        throw(ArgumentError("Flavor $flavor_symbol is not one of the valid flavors $valid_flavors"))
+    end
+
+    func = SaturationSpecificHumidityKernelFunction(flavor,
+                                                    model.microphysics,
+                                                    model.microphysical_fields,
+                                                    model.specific_moisture,
+                                                    model.temperature,
+                                                    model.formulation.reference_state,
+                                                    model.thermodynamic_constants)
+
+    return KernelFunctionOperation{Center, Center, Center}(func, model.grid)
+end
+
+@inline function saturation_total_specific_moisture(T, pᵣ, constants, equil)
+    surface = equilibrated_surface(equil, T)
+    pᵛ⁺ = saturation_vapor_pressure(T, constants, surface)
+    Rᵈ = dry_air_gas_constant(constants)
+    Rᵛ = vapor_gas_constant(constants)
+    δᵈᵛ = Rᵈ / Rᵛ - 1
+    return pᵛ⁺ / (pᵣ + δᵈᵛ * pᵛ⁺)
+end
+
+const AdjustmentSH = SaturationSpecificHumidityKernelFunction{<:SaturationAdjustment}
+
+function (d::AdjustmentSH)(i, j, k, grid)
+    @inbounds begin
+        pᵣ = d.reference_state.pressure[i, j, k]
+        ρᵣ = d.reference_state.density[i, j, k]
+        T = d.temperature[i, j, k]
+    end
+
+    constants = d.thermodynamic_constants
+    equil = d.microphysics.equilibrium
+
+    if d.flavor isa Prognostic
+        qᵗ = @inbounds d.specific_moisture[i, j, k]
+        q = compute_moisture_fractions(i, j, k, grid, d.microphysics, ρᵣ, qᵗ, d.microphysical_fields)
+        ρ = density(pᵣ, T, q, constants)
+        surface = equilibrated_surface(equil, T)
+        return saturation_specific_humidity(T, ρ, constants, surface)
+
+    elseif d.flavor isa Equilibrium
+        qᵗ = @inbounds d.specific_moisture[i, j, k]
+        return equilibrium_saturation_specific_humidity(T, pᵣ, qᵗ, constants, equil)
+
+    elseif d.flavor isa TotalMoisture
+        return saturation_total_specific_moisture(T, pᵣ, constants, equil)
+
+    end
+end
+
+SaturationSpecificHumidityField(model, flavor_symbol=:prognostic) = Field(SaturationSpecificHumidity(model, flavor_symbol))
```

Comment:
I'm confused, doesn't this method override https://github.com/NumericalEarth/Breeze.jl/blob/235feb4a777cce462e206378be8ed1086e4a75f8/src/Microphysics/microphysics_diagnostics.jl#L87?

Reply:
Removing this method would make all tests pass.

Reply:
i thnk its a mistake. I'm sorry Aqua 🥺 but also thank you